code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -1743,6 +1744,71 @@ type readMarkersCount struct {
 	ReadMarkersCount int64 `json:"read_markers_count"`
 }
 
+type readMarkerResponse struct {
+	ID			  uint
+	CreatedAt	  time.Time
+	UpdatedAt	  time.Time
+	Timestamp     int64   `json:"timestamp"`
+	ReadCounter   int64   `json:"read_counter"`
+	ReadSize      float64 `json:"read_size"`
+	Signature     string  `json:"signature"`
+	PayerID       string  `json:"payer_id"`
+	AuthTicket    string  `json:"auth_ticket"`   //used in readmarkers
+	BlockNumber   int64   `json:"block_number"` //used in alloc_read_size
+	ClientID	  string  `json:"client_id"`
+	OwnerID	      string  `json:"owner_id"`
+	TransactionID string  `json:"transaction_id"`
+	AllocationID  string  `json:"allocation_id"`
+
+	// TODO: Decide which pieces of information are important to the response
+	// Client 		*event.User
+	// Owner		*event.User
+	// Allocation	*event.Allocation
+}
+
+func toReadMarkerResponse(rm event.ReadMarker) readMarkerResponse {
+	return readMarkerResponse{
+		ID: rm.ID,
+		CreatedAt: rm.CreatedAt,
+		UpdatedAt: rm.UpdatedAt,
+		Timestamp: rm.Timestamp,
+		ReadCounter: rm.ReadCounter,
+		ReadSize: rm.ReadSize,
+		Signature: rm.Signature,
+		PayerID: rm.PayerID,
+		AuthTicket: rm.AuthTicket,
+		BlockNumber: rm.BlockNumber,
+		ClientID: rm.ClientID,
+		OwnerID: rm.OwnerID,
+		TransactionID: rm.TransactionID,
+		AllocationID: rm.AllocationID,
+		
+		// Client: &event.User{
+		// 	// ID: rm.User.ID,
+		// 	UserID: rm.User.UserID,
+		// 	Balance: rm.User.Balance,
+		// 	Change: rm.User.Change,
+		// 	Round: rm.User.Round,
+		// 	Nonce: rm.User.Nonce,
+		// },
+
+		// Owner: &event.User{
+		// 	// ID: rm.User.ID,
+		// 	UserID: rm.User.UserID,
+		// 	Balance: rm.User.Balance,
+		// 	Change: rm.User.Change,
+		// 	Round: rm.User.Round,
+		// 	Nonce: rm.User.Nonce,
+		// },
+
+		// Allocation: &event.Allocation{
+		// 	AllocationID: rm.Allocation.AllocationID,
+		// 	AllocationName: rm.Allocation.AllocationName,
+		// 	TransactionID: rm.Allocation.TransactionID,
+		// }
```

Comment:
I understand clearly that this is not a safe thing to do. However, ReadMarkers and WriteMarkers were always returned as "null" no matter what data in the DB, and that was the main issue.

This is what happens in all the other API endpoins for the txn that returns a collection of them. I tried to resolve the issue but in vain, looks like it's a problem with Gorm; but since no body is complaining about it, we can leave it as is for now till we have an open issue (not being lazy but since no body is having problem with it, this means RMs and WMs are not used by the clients of the API consuming these endpoints).

What I did here is that I fixed this behaviour for the endpoint returning a single TXN by hash, and that's by returning only 1 level of relationship from the txn (which is the txn itself, same as the struct, and the non-relational fields of it's direct children which are the RMs and WMs related to it). I want to stress on something that this very endpoint always returned null for RMs and WMs.